agent-sdk/examples/categories/chatbot/ai-chatbot-app/src/app/Chatbot/index.tsx

```diff
@@ -0,0 +1,112 @@
+import ReactMarkdown from 'react-markdown';
+import { useState, useRef, useEffect } from 'react';
+import {
+  ChatToggleButton,
+  ChatWindow,
+  ChatHeader,
+  ChatMessages,
+  MessageContainer,
+  MessageBubble,
+  MarkdownMessage,
+  ChatInputContainer,
+  ChatInput,
+} from './styles';
+
+export function Chatbot(): JSX.Element {
+  const [thread, setThread] = useState<{ role: string; content: string }[]>([]);
+  const [input, setInput] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const bottomRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (bottomRef.current) {
+      bottomRef.current.scrollIntoView({ behavior: 'smooth' });
+    }
+  }, [thread, loading]);
+
+  const sendMessage = async () => {
```

Comment:
race condition can occur
check is_loading,

if loading, make it fail, and return error